src/EnergyPlus/AirflowNetworkBalanceManager.cc

```diff
@@ -1209,9 +1220,37 @@ namespace AirflowNetworkBalanceManager {
 			// Wind coefficient == Surface-Average does not need inputs of external nodes
 			CurrentModuleObject = "AirflowNetwork:MultiZone:ExternalNode";
 			AirflowNetworkNumOfExtNode = GetNumObjectsFound( CurrentModuleObject );
+			if ( AnyLocalEnvironmentsInModel ) {
+ 				CurrentModuleObject = "OutdoorAir:Node";
+ 				AirflowNetworkNumOfOutAirNode = GetNumObjectsFound( CurrentModuleObject );
+ 				AirflowNetworkNumOfExtNode += AirflowNetworkNumOfOutAirNode;
+ 			}
 			if ( AirflowNetworkNumOfExtNode > 0 ) {
 				MultizoneExternalNodeData.allocate( AirflowNetworkNumOfExtNode );
-				for ( i = 1; i <= AirflowNetworkNumOfExtNode; ++i ) {
+				if ( AnyLocalEnvironmentsInModel ){
+ 					CurrentModuleObject = "OutdoorAir:Node";
+ 					for ( i = AirflowNetworkNumOfExtNode - AirflowNetworkNumOfOutAirNode + 1; i <= AirflowNetworkNumOfExtNode; ++i ) {
+ 						GetObjectItem( CurrentModuleObject, i - ( AirflowNetworkNumOfExtNode - AirflowNetworkNumOfOutAirNode ), Alphas, NumAlphas, Numbers, NumNumbers, IOStatus, lNumericBlanks, lAlphaBlanks, cAlphaFields, cNumericFields );
```

Comment:
@XuanLuoLBL I think it may be better to place this section after reading CurrentModuleObject = "AirflowNetwork:MultiZone:ExternalNode"; Here is an order I would like to see

Read AirflowNetwork:MultiZone:ExternalNode to keep original order
Add a new section to read "OutdoorAir:Node"